oneview_redfish_toolkit/blueprints/computer_system.py

```diff
@@ -103,6 +103,81 @@ def get_computer_system(uuid):
         return abort(status.HTTP_500_INTERNAL_SERVER_ERROR)
 
 
+@computer_system.route("/redfish/v1/Systems/<uuid>/"
+                       "Actions/ComputerSystem.Reset", methods=["POST"])
+def change_power_state(uuid):
+    """Change the Oneview power state for a specific Server hardware.
+
+        Return ResetType Computer System redfish JSON for a
+        given server hardware UUID.
+        Logs exception of any error and return abort.
+
+        Returns:
+            JSON: Redfish JSON with ComputerSystem ResetType.
+
+        Exceptions:
+            HPOneViewException: When some OneView resource was not found.
+            return abort(404)
+
+            OneViewRedfishError: When occur a power state mapping error.
+            return abort(400)
+
+            Exception: Unexpected error.
+            return abort(500)
+    """
+
+    reset_type = request.form["ResetType"]
```

Comment:
Can this fail? What happens if someone sends it without the payload or even with a wrong name in the parameter?

Reply:
Fixed.